### ISSUE TITLE
sfpowerkit:source:profile:reconcile -s fails with error: Cannot read property 'getUsername' of undefined

### ISSUE THREAD

**Issue by busybox0** [Accenture/sfpowerkit#637]
**Describe the bug**
`sfdx sfpowerkit:source:profile:reconcile -s` failing with error:

```
-------------------------------------------------------------------------------------------
sfpowerkit  -- The DX@Scale Developer Toolkit -Version:4.1.5 -Release:February 22
-------------------------------------------------------------------------------------------
INFO [2022-02-16 11:29:54.769 +0000] (sfpowerkit): Profiles Found in Entire Drirectory 24
INFO [2022-02-16 11:29:54.769 +0000] (sfpowerkit): Initiated Profile reconcile thread :1  with a chunk of 1 profiles
INFO [2022-02-16 11:29:54.769 +0000] (sfpowerkit): Profiles queued in thread :1 :
INFO [2022-02-16 11:29:54.769 +0000] (sfpowerkit): ["C:\\xx\\force-app\\main\\default\\profiles\\Admin.profile-meta.xml"]
ERROR [2022-02-16 11:29:54.770 +0000] (sfpowerkit): Cannot read property 'getUsername' of undefined
    err: {
      "type": "TypeError",
      "message": "Cannot read property 'getUsername' of undefined",
      "stack":
          TypeError: Cannot read property 'getUsername' of undefined
              at C:\Users\x\AppData\Local\sfdx\node_modules\sfpowerkit\src\impl\source\profiles\profileReconcile.ts:100:33
              at new Promise (<anonymous>)
              at ProfileReconcile.runWorkers (C:\Users\x\AppData\Local\sfdx\node_modules\sfpowerkit\src\impl\source\profiles\profileReconcile.ts:73:25)
              at ProfileReconcile.<anonymous> (C:\Users\x\AppData\Local\sfdx\node_modules\sfpowerkit\src\impl\source\profiles\profileReconcile.ts:58:18)
              at Generator.next (<anonymous>)
              at fulfilled (C:\Users\x\AppData\Local\sfdx\node_modules\sfpowerkit\lib\impl\source\profiles\profileReconcile.js:24:58)

```

**Expected behavior**
Version 4.0.3 runs fine without any problems. 4.1.3 to 4.1.5 are having the error.

**Desktop:**
- OS: Windows 10
- Node Version 14.17.0; 14.18.2
- Version of sfpowerkit 4.1.5
- sfdx-cli/7.137.1 win32-x64 node-v14.18.2

**Comment by azlam-abdulsalam** [comment]
Thanks for raising the issue @busybox0 . Looking into it

**Comment by sumanth-bolledla-acn** [comment]
I'm able to reproduce the issue as well. Seems like the "-s" flag is expecting the target username value to be specified instead of skipping the condition.
<img width="687" alt="image" src="https://user-images.githubusercontent.com/92749436/154339957-10323442-db3f-40e7-96d2-3c99e43a7d91.png">

**Comment by busybox0** [comment]
Azlam, thanks for quick turnaround. 
When it will be proper to expect the bug fix in a versioned release ?

**Comment by azlam-abdulsalam** [comment]
@busybox0  @sumanth-bolledla-acn  you can try 4.2.1 now, it will hit Prod in few days

**Comment by azlam-abdulsalam** [comment]
@jonny-harte  You can install it using sfdx plugins: install sfpowerkit@4.2.1. We will be promoting this next week

**Comment by vuha-acn** [comment]
@jonny-harte  - Can you confirm that the destination org has the metadata pushed already and not a clean or? If it doesn't contain the metadata, the reconcile will strip out the profile access rights from the profile files in order to enable deployment validation.  If you have successfully done this before in prior testing, then we can review the log files to confirm if there are any additional issues.

**Comment by jonny-harte** [comment]
Hi @vuha-acn, yes all the metadata is in my current authorised Org. When I retrieve, I get the full profile and then reconciling removes everything.

**Comment by jonny-harte** [comment]
@vuha-acn where do I specify that? Im just running the 2 commands below from the terminal within VSCode, which has my current working sandbox as the authorised Org. I want to capture all the profile updates in my sandbox and deploy them to production along with everything else in my pull request (Apex, flows etc.).

When I do a retrieve it captures a lot of extra metadata that I don't want and thought reconcile would strip it down and make it easier to deploy to production?

```
sfdx sfpowerkit:source:profile:retrieve --apiversion=54.0 --folder force-app --loglevel trace
sfdx sfpowerkit:source:profile:reconcile --folder force-app --sourceonly --loglevel trace
```

**Comment by vuha-acn** [comment]
@jonny-harte  - For the [command](https://github.com/Accenture/sfpowerkit#sfpowerkitsourceprofileretrieve), there is a -u, `--targetusername=targetusername` that identifies the source org you are retrieving from.  Then, when you do the reconcile, there is the same `-u, --targetorg=targetorg ` you run against.  I typically run it against the org not the "force-app" package folder.  Try the reconcile against the org instead of the force-app.  You can have a read of my [article](https://www.linkedin.com/pulse/version-controlling-profiles-why-makes-sense-deployments-vu-ha) here as well to get a broader view of how I typically use the commands.

**Comment by azlam-abdulsalam** [comment]
Its not fixed properly, as evident from #655

**Comment by azlam-abdulsalam** [comment]
This is fixed in 4.2.4 and part of April 22 release
